logisland-components/logisland-processors/logisland-processor-common/src/main/java/com/hurence/logisland/processor/EncryptField.java

```diff
@@ -0,0 +1,417 @@
+import com.hurence.logisland.annotation.documentation.CapabilityDescription;
+import com.hurence.logisland.annotation.documentation.ExtraDetailFile;
+import com.hurence.logisland.component.InitializationException;
+import com.hurence.logisland.component.PropertyDescriptor;
+import com.hurence.logisland.processor.encryption.*;
+
+import com.hurence.logisland.record.Field;
+import com.hurence.logisland.record.FieldType;
+import com.hurence.logisland.record.Record;
+import com.hurence.logisland.validator.StandardValidators;
+import com.hurence.logisland.validator.ValidationContext;
+import com.hurence.logisland.validator.ValidationResult;
+
+
+import javax.crypto.NoSuchPaddingException;
+import java.io.*;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import java.util.*;
+@ExtraDetailFile("./details/common-processors/EncryptField-Detail.rst")
+@CapabilityDescription("This is a processor that is used to encrypt or decrypt one or many fields of any type of a given Record mapping")
+@DynamicProperty(name = "field to encrypt",
+        supportsExpressionLanguage = true,
+        value = "a default value",
+        description = "encrypt the field value")
+public class EncryptField extends AbstractProcessor {
+
+
+    private static final long serialVersionUID = -270933070438408174L;
+
+    public static final String ENCRYPT_MODE = "Encrypt";
+    public static final String DECRYPT_MODE = "Decrypt";
+    public static final String AES = "AES";
+    public static final String AES_CBC_PKCS5 = "AES/CBC/PKCS5Padding";
+    public static final String AES_CBC_NoPAD = "AES/CBC/NoPadding";
+    public static final String AES_ECB_NoPAD = "AES/ECB/NoPadding";
+    public static final String AES_ECB_PKCS5 = "AES/ECB/PKCS5Padding";
+    public static final String DES = "DES";
+    public static final String DES_CBC_NoPAD = "DES/CBC/NoPadding";
+    public static final String DES_CBC_PKCS5 = "DES/CBC/PKCS5Padding";
+    public static final String DES_ECB_NoPAD = "DES/ECB/NoPadding";
+    public static final String DES_ECB_PKCS5 = "DES/ECB/PKCS5Padding";
+    public static final String DESede = "DESede";
+    public static final String DESede_CBC_NoPAD = "DESede/CBC/NoPadding";
+    public static final String DESede_CBC_PKCS5 = "DESede/CBC/PKCS5Padding";
+    public static final String DESede_ECB_NoPAD = "DESede/ECB/NoPadding";
+    public static final String DESede_ECB_PKCS5 = "DESede/ECB/PKCS5Padding";
+    public static final String RSA_ECB_PKCS5 = "RSA/ECB/PKCS1Padding";
+    public static final String RSA_ECB_OAE1 = "RSA/ECB/OAEPWithSHA-1AndMGF1Padding";
+    public static final String RSA_ECB_OAE256 = "RSA/ECB/OAEPWithSHA-256AndMGF1Padding";
+    public static final String RSA = "RSA";
+
+
+    public static final PropertyDescriptor MODE = new PropertyDescriptor.Builder()
+            .name("Mode")
```

Comment:
name property should be lower case as for all other processors.
description is not explicit.

please also don't use short name such as "algo" use "alogrithm" instead

this comment stands for all property descriptor in this processor